### ISSUE TITLE
need for celluloid socket in the client

### ISSUE THREAD

**Issue by camallen** [zooniverse/Cellect#45]
Is there still a need to run the client with a Celluloid::IO::TCPSocket, https://github.com/zooniverse/Cellect/blob/master/lib/cellect/client/connection.rb#L62 ?

**Comment by parrish** [comment]
It's not essential, but anytime I/O can be non-blocking is certainly beneficial.

Is it causing problems?

**Comment by camallen** [comment]
Not really no, just verbosity in the specs (cleaned up [here](https://github.com/zooniverse/Panoptes/commit/55304a5e24f54124e0de85b543090e9b297a3baa)) and I was wondering if this would be impacted by the GIL for MRI anyway.

**Comment by parrish** [comment]
Actually, those lines in the specs are for setting up (and quieting the logs from) celluloid. The specs will need that ceremony regardless of the I/O library used since the client's zookeeper node set is a celluloid actor.

Insofar as the GIL goes, it's pretty irrelevant to this.  Using a non-blocking connection just allows the app to not block waiting on the Cellect API to respond.  In other words, you can call most of the API actions without any performance impact on the rest of the app.

That said, Cellect asynchronously executes most of those actions, so the response times shouldn't be too long anyway -- I think most are < 10ms.

Long story short, unless we decide to remove celluloid from the client (which is certainly worth considering), we probably shouldn't remove non-blocking I/O from it unless it's causing problems.
